backend/dataall/base/utils/naming_convention.py

```diff
@@ -4,10 +4,21 @@
 
 
 class NamingConventionPattern(Enum):
-    S3 = {'regex': '[^a-zA-Z0-9-]', 'separator': '-', 'max_length': 63}
+    S3 = {
+        'regex': '[^a-zA-Z0-9-]',
+        'separator': '-',
+        'max_length': 63,
+        'valid_external_regex': '(?!(^xn--|.+-s3alias$))^[a-z0-9][a-z0-9-]{1,61}[a-z0-9]$',
+    }
+    KMS = {'regex': '[a-zA-Z0-9_-]+$', 'separator': '-', 'max_length': 63, 'valid_external_regex': '[a-zA-Z0-9_-]+$'}
```

Comment:
last thing - since `regex` is as `regex pattern for disallowed characters` in `def slugify(...)` I think best we remove the `regex` for `KMS` Pattern (or change to be the negation of what it is now)

Reply:
++ need to make `'valid_external_regex': '^[a-zA-Z0-9_-]+$'` to ensure it starts and ends with